main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -2292,18 +2304,8 @@ static void updateOfflineBox(final View view, final Geocache cache, final Resour
         final ImageButton offlineEdit = view.findViewById(R.id.offline_edit);
 
         // check if hint is available and set onClickListener and hint button visibility accordingly
-        boolean hintButtonEnabled = false;
-        if (null != showHintClickListener) {
-            final String hint = cache.getHint();
-            final boolean hintGiven = !StringUtils.isEmpty(hint);
-            final String personalNote = cache.getPersonalNote();
-            final boolean personalNoteGiven = !StringUtils.isEmpty(personalNote);
-            if (hintGiven || personalNoteGiven) {
-                hintButtonEnabled = true;
-                final TextView offlineHintText = view.findViewById(R.id.offline_hint_text);
-                offlineHintText.setText((hintGiven ? hint + (personalNoteGiven ? "\r\n" : "") : "") + (personalNoteGiven ? personalNote : ""));
-            }
-        }
+        final boolean hintButtonEnabled = null != showHintClickListener && setOfflineHintText(view.findViewById(R.id.offline_hint_text), cache.getHint(), cache.getPersonalNote());
```

Comment:
This is not easy to read.
How about to add showHintClickListener as extra parameter?

Reply:
updated accordingly